contexts/credentials-v1.jsonld

```diff
@@ -2,102 +2,137 @@
   "@context": {
     "@version": 1.1,
     "@protected": true,
-
     "id": "@id",
     "type": "@type",
-
     "VerifiableCredential": {
       "@id": "https://www.w3.org/2018/credentials#VerifiableCredential",
       "@context": {
         "@version": 1.1,
         "@protected": true,
-
         "id": "@id",
         "type": "@type",
-
         "cred": "https://www.w3.org/2018/credentials#",
         "sec": "https://w3id.org/security#",
         "xsd": "http://www.w3.org/2001/XMLSchema#",
-
         "credentialSchema": {
           "@id": "cred:credentialSchema",
           "@type": "@id",
           "@context": {
             "@version": 1.1,
             "@protected": true,
-
             "id": "@id",
             "type": "@type",
-
             "cred": "https://www.w3.org/2018/credentials#",
-
             "JsonSchemaValidator2018": "cred:JsonSchemaValidator2018"
           }
         },
-        "credentialStatus": {"@id": "cred:credentialStatus", "@type": "@id"},
-        "credentialSubject": {"@id": "cred:credentialSubject", "@type": "@id"},
-        "evidence": {"@id": "cred:evidence", "@type": "@id"},
-        "expirationDate": {"@id": "cred:expirationDate", "@type": "xsd:dateTime"},
-        "holder": {"@id": "cred:holder", "@type": "@id"},
-        "issued": {"@id": "cred:issued", "@type": "xsd:dateTime"},
-        "issuer": {"@id": "cred:issuer", "@type": "@id"},
-        "issuanceDate": {"@id": "cred:issuanceDate", "@type": "xsd:dateTime"},
-        "proof": {"@id": "sec:proof", "@type": "@id", "@container": "@graph"},
+        "credentialStatus": {
```

Comment:
The changes in this file introduces new line characters that are breaking changes to versioned JSON-LD Contexts. The addition of these bytes will cause systems that depend on the same hash of the context to fail.

Please resolve.

1. Restore the context to the version published here `https://www.w3.org/2018/credentials/v1`.
2. Write a test case that fetches the context found at the aforementioned url matches the context found in the repo by comparing their digest values.

Reply:
@gannan08 - I think the canonical context you're referring to, `https://www.w3.org/2018/credentials/v1`, is hosted off of https://github.com/w3c/vc-data-model/blob/main/contexts/credentials/v1, and not this repo.
This repo is just for NPM usage; nobody is checking hashes of it.

Reply:
> @gannan08 - I think the canonical context you're referring to, `https://www.w3.org/2018/credentials/v1`, is hosted off of https://github.com/w3c/vc-data-model/blob/main/contexts/credentials/v1, and not this repo.

Correct. The context found when resolving the url, is not being served by this repo. That's fine. The context found when resolving the url, does not match the version in this repo byte-for-byte. This is not fine.

> nobody is checking hashes of it.

It has been discussed in W3C WGs that the integrity JSON-LD Contexts is vital. From a security perspective, it is critical that the bytes in the context used here match the bytes one would consume if fetched from the context's url.

Is the security around contexts a concern you share?

Reply:
> Is the security around contexts a concern you share?

Absolutely. And I've long advocated (both at W3C and inside Digital Bazaar) that we need to standardize hashlinks for contexts, and built it into our jsonld.js infrastructure.
And I very much appreciate that you're taking a very careful approach to security here!

In the case of this particular repo, I don't think it applies, though. Mostly because - the generated `.jsonld` file in this repo is  actually redundant -- I added it only because it's part of our general context repo template, and I'll be removing it from this repo shortly. It's there in the general repo template because for many other contexts, the canonical context is hosted off of that same repo, from github pages, off of that .jsonld file. That is not the case for this repo though. This repo only exports the context as an NPM object. (Not a string.) And as such, it's not even possible to do byte-for-byte comparison with the `vc-data-model` context.

In general, I don't think hash checking of contexts, for security, should ever apply to whitespace characters. When the discussion is brought up again in the W3C community, I will advocate for only comparing canonicalized representations (basically, JCS canonicalization + a `JSON.stringify()`).

Does that make sense?